### PR TITLE
homie-device: Use new AsyncClient.

### PR DESCRIPTION
**Pull request by qwandor-google**

### CODE REVIEW

File: homie-device/src/lib.rs [hunk 1]
```diff
@@ -489,29 +488,23 @@ impl DevicePublisher {
...
-        Ok(())
+        self.client
+            .publish(topic, QoS::AtLeastOnce, true, value)
+            .await
     }
 
     async fn disconnect(&self) -> Result<(), ClientError> {
-        self.requests_tx.send(Request::Disconnect).await?;
-        Ok(())
+        self.client.disconnect().await
```
Comment (alsuren) [inline homie-device/src/lib.rs:497]: Maybe just delete this method and use the one on client instead?

File: publish-mqtt/src/main.rs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#![type_length_limit = "1129369"]
```
Comment (alsuren) [inline publish-mqtt/src/main.rs:1]: :-(

**Review by alsuren — APPROVED** [review]
This is a lot easier to read than before.

File: homie-device/src/lib.rs (hunk 1 above)
Comment (qwandor-google) [reply homie-device/src/lib.rs]: Done.

**Pull request merged**